x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -234,6 +236,18 @@
         Setting.Property.NotCopyableOnResize
     );
 
+    /**
+     * Index setting used to indicate if the snapshot that is mounted as an index should be deleted when the index is deleted. This setting
+     * is only set for indices mounted in clusters on or after 8.0.0. Once set this setting cannot be updated.
+     */
+    public static final Setting<Boolean> DELETE_SEARCHABLE_SNAPSHOT_ON_INDEX_DELETION = Setting.boolSetting(
+        SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION,
+        false,
+        Setting.Property.IndexScope,
+        Setting.Property.PrivateIndex,
+        Setting.Property.NotCopyableOnResize
```

Comment:
Can we shrink/split/clone searchable snapshots and what is the result (a regular index or a searchable snapshot)? Not really a comment to this PR, just a question :slightly_smiling_face:

Reply:
I expect shrink & split actions to fail because if the number of shards is changed it won't match the number of shards in the snapshot. I'll take a look at this separately, we should prevent this I think. Clone should work but I'll verify this too.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/75227 to prevent resize actions on searchable snapshot indices.

Reply:
Cloning is supported since #56595